Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs

```diff
@@ -89,6 +90,20 @@ public override void ViewDidDisappear(bool animated)
 
 			_appeared = false;
 			PageController.SendDisappearing();
+
+			var navigationPage = Element.Parent as NavigationPage;
+			if (navigationPage == null)
+				return;
+
+			int managedStackCount = navigationPage.Navigation.NavigationStack.Count;
+			IVisualElementRenderer renderer = Platform.GetRenderer(navigationPage);
+
+			var navigationController = renderer as UINavigationController;
+			if (navigationController != null && managedStackCount > navigationController.ViewControllers.Length)
+			{
+				Task<Page> task = ((INavigationPageController)renderer.Element).PopAsyncInner(animated, true);
```

Comment:
The fix should be on NavigationPage and not here.. I think the code in NavigationPAge is not good and we should fix it.. this looks more like a patch  ..